test/testfiles/test.xliff

```diff
@@ -16,6 +16,11 @@
         <target>bebe bebe</target>
         <note annotates="source">come &amp; enjoy it with us</note>
       </trans-unit>
+      <trans-unit id="3" resname="ohhuzzah" restype="string" datatype="plaintext">
+        <source x-key="oh">baby baby</source>
+        <target>oh bebe bebe</target>
+        <note annotates="source">come &amp; enjoy it with them</note>
+      </trans-unit>
```

Comment:
For this test, the x-key replaces the resname, right? You should have a test where the trans-unit has no resname attribute on it, as well as this test, and make sure that the resname turns out correctly.

Reply:
I've added another `trans-unit`  which doesn't have `rename` and `datatype`